dotCMS/src/main/java/com/dotcms/security/apps/AppsUtil.java

```diff
@@ -0,0 +1,150 @@
+     * @param object
+     * @return
+     * @throws DotDataException
+     */
+    static char[] toJsonAsChars(final AppSecrets object) throws DotDataException {
+        try {
+            final byte [] bytes = mapper.writeValueAsBytes(object);
+            return bytesToCharArrayUTF(bytes);
+        } catch (IOException e) {
+            throw new DotDataException(e);
+        }
+    }
+
+    /**
+     * Takes a char array representation of a json secret and generates the domain object.
+     * No strings are created in the transformation process.
+     * @param chars
+     * @return The Secrets domain model.
+     * @throws DotDataException
+     */
+    static AppSecrets readJson(final char[] chars) throws DotDataException {
+        try {
+            final byte [] bytes = charsToBytesUTF(chars);
+            return mapper.readValue(bytes, AppSecrets.class);
+        } catch (IOException e) {
+            throw new DotDataException(e);
+        }
+    }
+
+    /**
+     * This method takes a byte array and converts its contents into a char array
+     * No String middle man is created.
+     * https://stackoverflow.com/questions/8881291/why-is-char-preferred-over-string-for-passwords
+     * @param bytes
+     * @return char array
+     */
+    static char[] bytesToCharArrayUTF(final byte[] bytes) throws IOException {
+        final List<Integer> integers = new ArrayList<>(bytes.length);
+        try (final BufferedReader reader = new BufferedReader(
+                new InputStreamReader(new ByteArrayInputStream(bytes), StandardCharsets.UTF_8))) {
+
+            int chr;
+            while ((chr = reader.read()) != -1) {
+                integers.add(chr);
+            }
+        }
+        return ArrayUtils.toPrimitive(
+                integers.stream().map(value -> (char) value.intValue()).toArray(Character[]::new));
+    }
+
+    /**
+     * This method takes a char array and converts its contents into a byte array No String middle
+     * man is created. https://stackoverflow.com/questions/8881291/why-is-char-preferred-over-string-for-passwords
+     * @param chars input
+     * @return byte array
+     */
+    static byte[] charsToBytesUTF(final char[] chars) throws IOException {
+        final CharSequence sequence = java.nio.CharBuffer.wrap(chars);
+        return ByteStreams
+                .toByteArray(new CharSequenceInputStream(sequence, StandardCharsets.UTF_8));
+    }
+
+
+    /**
+     * Encrypt variant of the function of the same name located in The Encryptor util class.
+     * The main difference is that this ones does not use a string in the middle. It directly takes a char array.
+     * And doesn't use a string internally.
+     * @see Encryptor#encrypt(Key, String)
+     * @param key security Key
+     * @param chars
+     * @return encrypted text as a char array
+     * @throws EncryptorException
+     */
+    static char[] encrypt(final Key key, final char[] chars)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5329810)